src/containers/PageContainer/index.tsx

```diff
@@ -1,239 +1,108 @@
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.METAMASK ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.COINBASE ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.PORTIS ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.FORTMATIC) &&
+			Object.values(WALLETS).includes(
+				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) as WALLETS,
+			) &&
 			!active &&
 			triedEagerConnect
 		) {
-			setTimeout(async () => {
-				localStorage.removeItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET);
-			}, 2000);
+			localStorage.removeItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET);
 		}
+
 		if (triedEagerConnect)
-			addNotification(active ? 'Wallet connected.' : 'Wallet disconnected.');
+			addNotification(
+				active
+					? WALLET_NOTIFICATIONS.CONNECTED
+					: WALLET_NOTIFICATIONS.DISCONNECTED,
+			);
 
 		// Check if we need to close a modal
 		if (modal === Modal.Transfer || modal === Modal.Mint) {
 			closeModal();
 		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [active]);
+	}, [active, modal, triedEagerConnect, addNotification, closeModal]);
 
-	useEffect(() => {
-		// TODO: Clean this whole hook up
-		if (znsDomain) {
-			// Set the domain data for table view
-			setHasLoaded(true);
-		}
-		window.scrollTo({
-			top: -1000,
-			behavior: 'smooth',
-		});
-	}, [znsDomain, hasLoaded]);
-
-	/////////////////////
-	// React Fragments //
-	/////////////////////
-
-	const modals = () => (
-		<>
-			{/* Overlays */}
-			<NotificationDrawer />
-			<ProfileModal />
-			<Overlay style={{ zIndex: 3 }} open={isSearching} onClose={() => {}} />
-			{modal === Modal.Wallet && (
-				<Overlay centered open={modal === Modal.Wallet} onClose={closeModal}>
-					<ConnectToWallet onConnect={closeModal} />
-				</Overlay>
-			)}
-			{modal === Modal.Mint && (
-				<Overlay open onClose={closeModal}>
-					<MintNewNFT
-						onMint={closeModal}
-						domainName={domain}
-						domainId={znsDomain ? znsDomain.id : ''}
-						domainOwner={znsDomain ? znsDomain.owner.id : ''}
-						subdomains={
-							(znsDomain?.subdomains?.map(
-								(sub: any) => sub.name,
-							) as string[]) || []
-						}
-					/>
-				</Overlay>
-			)}
-		</>
-	);
+	/**
+	 * Life Cycles
+	 */
+	useUpdateEffect(handleChainSelect, [chainId]);
+	useUpdateEffect(handleForceBackHome, [znsDomain, loading, globalDomain.app]);
+	useUpdateEffect(handleWalletChanges, [active]);
+	useUpdateEffect(refetch, [minted, refetch, stakingFulFilled]);
 
 	return (
-		<>
-			{pageWidth > 1000 && modals()}
-
+		<ScrollToTop>
 			<div
-				className="page-spacing"
-				style={{
-					opacity: hasLoaded ? 1 : 0,
-					transition: 'opacity 0.2s ease-in-out',
-				}}
+				className={classnames(styles.PageContainer, {
+					[styles.isVisible]:
+						Boolean(znsDomain) ||
+						Boolean(prevZnsDomain) ||
+						Boolean(globalDomain.domain),
+				})}
```

Comment:
FYI you can use `classnames/bind` for this!

```js
import classnames from 'classnames/bind'
const cx = classnames.bind(styles)

// in element classNames prop
cx(styles.PageContainer, {
    isVisible: something that evaluates to a boolean
})
```

The above will conditionally apply the `ModuleClassName` class